src/golang/cmd/migrator/versions/000019_add_serialization_value_to_param_op/utils.go

```diff
@@ -0,0 +1,110 @@
+package _000019_add_serialization_value_to_param_op
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/aqueducthq/aqueduct/lib/collections/utils"
+	"github.com/aqueducthq/aqueduct/lib/database"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	serilization_type_key   = "serialization_type"
+	value_key               = "val"
+	spec_field              = "spec"
+	pythonExecutorPackage   = "aqueduct_executor"
+	typeInferencePythonPath = "migrators.parameter_val_type_inference_000019.main"
+)
+
+type Operator struct {
+	Id     uuid.UUID `db:"id" json:"id"`
+	OpSpec Spec      `db:"spec" json:"spec"`
+}
+
+func getAllOperators(
+	ctx context.Context,
+	db database.Database,
+) ([]Operator, error) {
+	query := "SELECT id, spec FROM operator;"
+
+	var response []Operator
+	err := db.Query(ctx, &response, query)
+	return response, err
+}
+
+func updateParamOperatorWithNewSpec(
+	ctx context.Context,
+	operator Operator,
+	db database.Database,
+) error {
+	param_val := operator.OpSpec.Param[value_key]
+	// Launch the Python job to infer the type of the parameter value
+	cmd := exec.Command(
+		"python3",
+		"-m",
+		fmt.Sprintf("%s.%s", pythonExecutorPackage, typeInferencePythonPath),
+		"--val",
+		param_val,
+	)
+	cmd.Env = os.Environ()
+
+	var outb, errb bytes.Buffer
+	cmd.Stdout = &outb
+	cmd.Stderr = &errb
+
+	err := cmd.Run()
+	if err != nil {
+		log.Errorf("Error running Python migration job. Stdout: %s, Stderr: %s.", outb.String(), errb.String())
+		return err
+	}
+
+	param_type := strings.TrimSuffix(outb.String(), "\n")
+	operator.OpSpec.Param[serilization_type_key] = param_type
+
+	// We also change the param value to be a base64 encoding
+	operator.OpSpec.Param[serilization_type_key] = param_type
+
+	base64val := base64.StdEncoding.EncodeToString([]byte(param_val))
+	operator.OpSpec.Param[value_key] = base64val
+
+	newParamSpec := &Spec{
+		Type:  operator.OpSpec.Type,
+		Param: operator.OpSpec.Param,
+	}
+
+	changes := map[string]interface{}{
+		spec_field: newParamSpec,
+	}
+
+	return utils.UpdateRecord(ctx, changes, "operator", "id", operator.Id, db)
+}
+
+func updateParamOperatorWithOldSpec(
+	ctx context.Context,
+	operator Operator,
+	db database.Database,
+) error {
+	// If we want to migrate back to old version we delete the serialization type in the spec
+	delete(operator.OpSpec.Param, serilization_type_key)
+	// Decode base64 value back to string
+	decodedVal, _ := base64.StdEncoding.DecodeString(operator.OpSpec.Param[value_key])
+	operator.OpSpec.Param[value_key] = string(decodedVal)
```

Comment:
I'm not sure if `string(decodedVal)` restores the value back to its original json-serialized form. If so this will imply utf-8 encoding and json serialization are the same, which I don't think it's true? Worst case we may need to do this reverse conversion on the Python side as well.